src/demos/add-charts/deckgl-overlay.js

```diff
@@ -27,63 +26,46 @@ const elevationRange = [0, 1000];
 
 export default class DeckGLOverlay extends Component {
 
-  static get defaultViewport() {
-    return {
-      longitude: -74,
-      latitude: 40.7,
-      zoom: 11,
-      maxZoom: 16,
-      pitch: 0,
-      bearing: 0
-    };
-  }
-
   _initialize(gl) {
     gl.enable(gl.DEPTH_TEST);
     gl.depthFunc(gl.LEQUAL);
   }
 
   render() {
-    const {viewport, data, onHover, settings} = this.props;
-
-    if (!data) {
+    if (!this.props.data) {
       return null;
     }
 
     const layers = [
-      !settings.showHexagon ? new ScatterplotLayer({
+      !this.props.showHexagon ? new ScatterplotLayer({
         id: 'scatterplot',
-        data,
         getPosition: d => d.position,
         getColor: d => d.pickup ? PICKUP_COLOR : DROPOFF_COLOR,
         getRadius: d => 1,
         opacity: 0.5,
         pickable: true,
-        onHover,
-        radiusScale: 30,
         radiusMinPixels: 0.25,
-        radiusMaxPixels: 30
+        radiusMaxPixels: 30,
+        ...this.props
```

Comment:
here I'm using ...this.props to avoid the destructuring above.

Reply:
yeah I was on the fence here. I agree we should list the properties and associate them with the respective prop ie
 
data: this.props.data, 
etc.